devito/function.py

```diff
@@ -815,6 +786,68 @@ def __indices_setup__(cls, **kwargs):
     def __shape_setup__(cls, **kwargs):
         return kwargs.get('shape', (kwargs.get('npoint'),))
 
+    def _arg_defaults(self, alias=None):
+        """
+        Returns a map of default argument values defined by this symbol.
+
+        :param alias: (Optional) name under which to store values.
+        """
+        key = alias or self
+        args = super(AbstractSparseFunction, self)._arg_defaults(alias=alias)
+        for child_name in self._child_functions:
+            child = getattr(self, child_name)
```

Comment:
why don't you just put the actual functions in ``_child_functions`` rather than their names?

Reply:
If I have a receiver symbol called `rec`, `self.coordinates.name` on it would give you `rec_coordinates`. If it is replaced by a symbol called `rec2`, the only way of accessing the coordinates `Function` on it is by doing `getattr(rec2, 'coordinates')` since (assuming `child = self.coordinates`) `getattr(rec2, child.name)` won't be found (the new symbol's coordinates function is automatically named `rec2_coordinates`.

Reply:
Which also means this should probably be `child = getattr(key, child_name)`. Changing that.

Reply:
This line: `args.update(child._arg_defaults(alias=getattr(key, child_name)))` will not work when storing the actual functions, only with the name, since the name is the only consistent way of accessing children across different `SparseFunction` instances.

Reply:
This is really odd, `child._arg_defaults` should pick up the name itself and all `arg_defaults` take an object as alias, not a name.

Reply:
sorry, I didn't see `_child_function` was a class attribute. Now it does make sense

Reply:
@mloubout there is no difference between what this line is doing and what is being done [here](https://github.com/opesci/devito/blob/master/devito/function.py#L1023), except here, it works for any arbitrarily named child attributes, not just `self.coordinates`. So, for a `PrecomputedSparseFunction`, this will execute:
`args.update(self.gridpoints._arg_defaults(alias=key.gridpoints)`. I believe I am reproducing the same behaviour as master, but in a more general way.